src/metabase/driver/postgres.clj

```diff
@@ -298,6 +303,70 @@
   [driver [_ arg]]
   (sql.qp/->honeysql driver [:percentile arg 0.5]))
 
+(defn- datetime-diff-helper [x y unit]
+  (case unit
+    (:year :day)
+    (hx/cast
+     :integer
+     (hsql/call
+      :extract
+      unit
+      (hsql/call
+       (case unit :year :age :day :-)
+       (hsql/call :date_trunc (hsql/raw "'day'") y)
+       (hsql/call :date_trunc (hsql/raw "'day'") x))))
+
+    :week
+    (hx// (datetime-diff-helper x y :day) 7)
+
+    :month
+    (hx/cast
+     :integer
+     (hx/+
+      (hx/* 12 (datetime-diff-helper x y :year))
+      (hsql/call
+       :extract
+       :month
+       (hsql/call
+        :age
+        (hsql/call :date_trunc (hsql/raw "'day'") y)
```

Comment:
```suggestion
        (hsql/call :date_trunc (hx/literal :day) y)
```

might be a little easier on the eyes

Reply:
Updated in https://github.com/metabase/metabase/commit/71f2e30f0df16fe84a9eb932096ffd7a75a3af7c